lib/teleterm/apiserver/handler/handler_auth.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2021 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handler
+
+import (
+	"context"
+
+	api "github.com/gravitational/teleport/lib/teleterm/api/protogen/golang/v1"
+
+	"github.com/gravitational/trace"
+)
+
+// Login logs in a user to a cluster
+func (s *Handler) Login(ctx context.Context, req *api.LoginRequest) (*api.EmptyResponse, error) {
+	cluster, err := s.DaemonService.ResolveCluster(req.ClusterUri)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	if req.Local != nil {
+		if err := cluster.LocalLogin(ctx, req.Local.User, req.Local.Password, req.Local.Token); err != nil {
```

Comment:
Currently, the TOTP is also handled by `client.SSHAgentLogin` where the prompt is skipped if token is part of the request. This is what `teleterm` uses to handle TOTPs.

In case of  `webauthn` or `utf`, it uses `SSHAgentMFALogin` where `PromptMFAChallenge` is called that awaits for a device to be activated. Both methods do not require CLI input.